### PR TITLE
Internal: Elements experiment [EDS-227]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

**Review by StyleShit — APPROVED** [review]

**Review by ManorHazaz — APPROVED** [review]

File: core/editor/editor-v2-experiments.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			'new_site' => [
+				'default_active' => true,
+				'minimum_installation_version' => '3.23.0',
+			],
+		] );
+
+		Plugin::$instance->experiments->add_feature( [
+			'name' => static::ELEMENTS,
+			'title' => esc_html__( 'Elements', 'elementor' ),
+			'description' => esc_html__( 'Enable the new elements.', 'elementor' ),
```
Comment (Omerisra6) [inline core/editor/editor-v2-experiments.php:43]: Is not the app bar is a dependency of the elements?

File: core/editor/loader/v2/editor-v2-loader.php [hunk 2]
```diff
@@ -15,19 +18,31 @@ class Editor_V2_Loader extends Editor_Base_Loader {
...
-		'editor-v1-adapters',
+		Editor_V2_Experiments::APP_BAR => [
+			'editor-app-bar',
+			'editor-documents',
+			'editor-panels',
+			'editor-responsive',
+			'editor-site-navigation',
+			'editor-v1-adapters',
+		],
+		Editor_V2_Experiments::ELEMENTS => [
```
Comment (Omerisra6) [inline core/editor/loader/v2/editor-v2-loader.php:32]: Why do we have to include the packages that are required for both experiments in both lists?
Isn't the app bar a dependency of the elements?

File: tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php [hunk 3]
```diff
@@ -14,49 +16,58 @@
...
 		// Act
-		$config_provider = Editor_Loader_Factory::create();
+		$loader = Editor_Loader_Factory::create();
 
 		// Assert
-		$this->assertInstanceOf( $expected_class, $config_provider );
+		$this->assertInstanceOf( $expected_class, $loader );
 	}
 
 	public function create_data_provider() {
```
Comment (Omerisra6) [inline tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php:54]: It feels weird to me, maybe we could split the `test_create` into two tests one for when the experiments are active and one for when they are inactive.

```php
/**
	 * @experiments Editor_V2_Experiments::all()
	 */	
	public function test_create_experiment_active( $experimemt ) {		
		Plugin::$instance->experiments->set_feature_default_state( $experiment, 				Experiments_Manager::STATE_ACTIVE);
		$loader = Editor_Loader_Factory::create();

		$this->assertInstanceOf( Editor_V2_Loader::class, $loader );
	}
```

**Review by Omerisra6 — CHANGES_REQUESTED** [review]

File: core/editor/editor-v2-experiments.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
+<?php
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply core/editor/editor-v2-experiments.php:1]: they rely on the same infra

File: core/editor/editor-v2-experiments.php (hunk 1 above)
Comment (StyleShit) [reply core/editor/editor-v2-experiments.php:43]: no

File: core/editor/loader/v2/editor-v2-loader.php (hunk 2 above)
Comment (StyleShit) [reply core/editor/loader/v2/editor-v2-loader.php:32]: no

File: tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php (hunk 3 above)
Comment (StyleShit) [reply tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php:54]: but I have multiple experiments

File: core/editor/editor-v2-experiments.php (hunk 1 above)
Comment (Omerisra6) [reply core/editor/editor-v2-experiments.php:43]: Isn't that the product requirement?

File: tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php (hunk 3 above)
Comment (StyleShit) [reply tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php:54]: IDK, that's the whole point of data providers... 🤷

Comment (Omerisra6) [reply tests/phpunit/elementor/core/editor/loader/test-editor-loader-factory.php:54]: So I don't like them...

File: core/editor/editor-v2-experiments.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (Omerisra6) [reply core/editor/editor-v2-experiments.php:1]: So maybe we should change the name of the module? I have an idea for a name but the "elements" are not a part of the v2 project they are the beginning of a different one.

File: core/editor/editor-v2-experiments.php (hunk 1 above)
Comment (StyleShit) [reply core/editor/editor-v2-experiments.php:43]: no

**Review by Omerisra6 — APPROVED** [review]

**Pull request merged**